Kernel/Devices/HID/PS2KeyboardDevice.cpp

```diff
@@ -86,7 +87,9 @@ void PS2KeyboardDevice::handle_irq(const RegisterState&)
 {
     // The controller will read the data and call irq_handle_byte_read
     // for the appropriate device
-    m_i8042_controller->irq_process_input_buffer(HIDDevice::Type::Keyboard);
+    g_io_work->queue([this]() {
+        m_i8042_controller->irq_process_input_buffer(HIDDevice::Type::Keyboard);
```

Comment:
I don't think that's going to work right. I think the 8i042's `irq_process_input_buffer` must be called from inside the IRQ handler...

Reply:
I think only the call to `ConsoleManagement::switch_to` needs to be called from the worker queue.

Reply:
Why? what is limiting us from handling input outside of the IRQ?

Reply:
> Why? what is limiting us from handling input outside of the IRQ?

EOI?

Reply:
The code in the `irq_` was meant to run inside the IRQ handler. But I'm no i8042 expert (I struggled with it too much), so maybe that's ok. But I thought the packets had to be read inside the IRQ handler.

Reply:
In fact it has a `VERIFY(Processor::current().in_irq());` right in the first line ;) If that isn't triggering then we have yet another bug ;)

Reply:
I changed that line, so it worked (it asserted) until I changed it. Maybe we should only let the switch to happen in an IO work queue then.

Reply:
Yeah I think only the call to `ConsoleManagement::switch_to`